python/tests/scripts/c1_grounded_basic.metta

```diff
@@ -100,21 +127,36 @@
     (* $x $x))
   (25))
 
-; Grounded symbols work with non-determinism based "reasoning"
-
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ; basic recursion
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 (= (fact $n) (if (== $n 0) 1 (* (fact (- $n 1)) $n)))
 !(assertEqual (fact 5) 120)
 
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+; Grounded symbols work with non-determinism based "reasoning"
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+; Uses the grounded call `bin`, which returns a list of
+; both possible bits (0, 1), to recursively construct all possible
+; binary lists of length `$n`, appending bits one by one.
 (= (gen $n)
    (if (> $n 0)
        (:: (bin) (gen (- $n 1)))
        nil))
+; Note: `::` is just a custom symbol, used here as a constructor,
+; but any other symbol can be used for this.
+
+; Calculates the sum of element-wise products between two lists
 (= (subsum nil nil) 0)
 (= (subsum (:: $x $xs) (:: $b $bs))
    (+ (* $x $b)
       (subsum $xs $bs)))
 
+; Non-determinism "reasoning":
+; Among all 3-bit binary lists, return the one whose `subsum`
+; with (:: 3 (:: 7 (:: 5 nil))) equals 8, or `nop` if not found
```

Comment:
JIC, nop is a grounded function that consumes its arguments and returns nothing.